docs/reference/java-interop.md

```diff
@@ -391,20 +410,71 @@ fun bar() {
 ``` java
 // Java
 new demo.Foo();
-demo.DemoPackage.bar();
+demo.ExampleKt.bar();
 ```
 
-For the root package (the one that's called a "default package" in Java), a class named `_DefaultPackage` is created.
+The name of the generated Java class can be changed using the `@JvmName` annotation:
+
+``` kotlin
+@file:JvmName("DemoUtils")
+
+package demo
+
+class Foo
+
+fun bar() {
+}
+
+```
+
+``` java
+// Java
+new demo.Foo();
+demo.DemoUtils.bar();
+```
+
+Having multiple files which have the same generated Java class name (the same package and the same name or the same
+@JvmName annotation) is normally an error. However, the compiler has the ability to generate a single Java facade
+class which has the specified name and contains all the declarations from all the files which have that name.
+To enable the generation of such a facade, use the @JvmMultifileClass annotation in all of the files.
+
+``` kotlin
+// oldutils.kt
+@file:JvmName("Utils")
+@file:JvmMultifileClass
```

Comment:
Do we want to promote `@[...]`  syntax? 
E.g., here you can write `@file:[JvmName("Utils") JvmMultifileClass]`.